TombEngine/Objects/TR3/Entity/tr3_compies.cpp

```diff
@@ -0,0 +1,330 @@
+			item->Timer++;
+
+			if (item->HitStatus && item->Timer > 200 && GetRandomControl() < COMPSOGNATHUS_ATTACK_CHANCE * 100 || (creature->Alerted && AI.zoneNumber == AI.enemyZone))
+			{
+					item->ItemFlags[1] = ATTACK_LARA;					
+			}
+
+			switch (item->Animation.ActiveState)
+			{
+			case COMPSOGNATHUS_STATE_IDLE:
+				creature->MaxTurn = COMPSOGNATHUS_STOP_TURN;
+				creature->Flags &= ~COMPSOGNATHUS_HIT_FLAG;
+				if (creature->Mood == MoodType::Attack)
+				{					
+					if (AI.ahead && AI.distance < COMPSOGNATHUS_ATTACK_RANGE * 4)
+					{
+						if (item->ItemFlags[1] == ATTACK_LARA)
+						{
+							if (GetRandomControl() < 0x4000)
+								item->Animation.TargetState = COMPSOGNATHUS_STATE_ATTACK;
+							else
+								item->Animation.TargetState = COMPSOGNATHUS_STATE_JUMP_ATTACK;
+						}
+						else
+							item->Animation.TargetState = COMPSOGNATHUS_STATE_IDLE;
+					}
+					else if (AI.distance > COMPSOGNATHUS_ATTACK_RANGE * (9 - 4 * target))
+						item->Animation.TargetState = COMPSOGNATHUS_STATE_RUN;
+				}
+				else if (creature->Mood == MoodType::Bored)
+				{					
+					if (AI.ahead && AI.distance < (COMPSOGNATHUS_ATTACK_RANGE * 3 ) && item->ItemFlags[1] == ATTACK_CADAVER)
+					{
+						if (GetRandomControl() < 0x4000)
+							item->Animation.TargetState = COMPSOGNATHUS_STATE_ATTACK;
+						else
+							item->Animation.TargetState = COMPSOGNATHUS_STATE_JUMP_ATTACK;	
+					}
+					else if (AI.distance > COMPSOGNATHUS_ATTACK_RANGE * 3)
+					{
+						item->Animation.TargetState = COMPSOGNATHUS_STATE_RUN;
+					}
+				}
+				else 
+				{
+					if (GetRandomControl() < COMPSOGNATHUS_JUMP_ATTACK_CHANCE)
+						item->Animation.TargetState = COMPSOGNATHUS_STATE_JUMP_ATTACK;
+					else
+						item->Animation.TargetState = COMPSOGNATHUS_STATE_RUN;
+				}
+
+				break;
+
+			case COMPSOGNATHUS_STATE_RUN:
+
+				creature->Flags &= ~COMPSOGNATHUS_HIT_FLAG;
+				creature->MaxTurn = COMPSOGNATHUS_RUN_TURN;
+
+				if (AI.angle < COMPSOGNATHUS_ATTACK_ANGLE && AI.angle > -COMPSOGNATHUS_ATTACK_ANGLE && AI.distance < COMPSOGNATHUS_ATTACK_RANGE * (9 - 4 * target))
+				{
+					item->Animation.TargetState = COMPSOGNATHUS_STATE_IDLE;
+
+				}
+
+				break;
+
+			case COMPSOGNATHUS_STATE_ATTACK:
+			case COMPSOGNATHUS_STATE_JUMP_ATTACK:
+
+				creature->MaxTurn = COMPSOGNATHUS_RUN_TURN;
+
+				if (!(creature->Flags & COMPSOGNATHUS_HIT_FLAG) && (item->TouchBits.Test(CompyAttackJoints) && item->ItemFlags[1] == ATTACK_LARA))
+				{
+					creature->Flags |= COMPSOGNATHUS_HIT_FLAG;
+
+					DoDamage(creature->Enemy, COMPSOGNATHUS_ATTACK_DAMAGE);
+					LaraItem->HitStatus = 1;
+					CreatureEffect(item, CompyBite, DoBloodSplat);
+				}
+				else if (!(creature->Flags & COMPSOGNATHUS_HIT_FLAG) && AI.distance < COMPSOGNATHUS_ATTACK_RANGE && AI.ahead && item->ItemFlags[1] == ATTACK_CADAVER)
+				{
+					creature->Flags |= COMPSOGNATHUS_HIT_FLAG;				
+					CreatureEffect(item, CompyBite, DoBloodSplat);
+				}
+
+				break;
+			}
+		}
+		tilt = angle >> 1;
```

Comment:
you can pass it directly to CreatureTilt if it's not changed anywhere.
`CreatureTilt(item, angle >> 1);`

Reply:
fixed